### PR TITLE
Curve Pools events

### PR DESCRIPTION
**Pull request by e1Ru1o**
Detect `RampA` & `StopRampA` events

Remarks:
- Agent core improved
- Test Improved
- Findings Improved

### CODE REVIEW

File: Curve-finance-suite/Address-Provider-Ownership-Transfer/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,16995 @@
+{
```
Comment (mauricio1802) [inline Curve-finance-suite/Address-Provider-Ownership-Transfer/package-lock.json:1]: I think this shouldn't be here

File: Curve-finance-suite/Pools-RampA/src/agent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  FindingSeverity,
+  Finding,
+  HandleTransaction,
+  createTransactionEvent,
+  TransactionEvent,
+} from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools";
+import agent from "./agent";
+import { when } from "jest-when";
+import { encodeParameters } from "nethermindeth-general-agents-module";
```
Comment (mauricio1802) [inline Curve-finance-suite/Pools-RampA/src/agent.spec.ts:12]: Shouldn't this be imported from `forta-agent-tools`?

File: Curve-finance-suite/Pools-RampA/src/agent.spec.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        "Custom(uint256, uint256)",
+        POOLS[1],
+        encodeParameters(["uint256", "uint256"], ["20", "10"]),
+      );
+
+    const findings: Finding[] = await handler(tx);
+    expect(findings).toStrictEqual([]);
+  });
+
+  it("should ignore report the events in Pools", async () => {
```
Comment (mauricio1802) [inline Curve-finance-suite/Pools-RampA/src/agent.spec.ts:97]: I think this should be rephrased.

**Review by mauricio1802 — CHANGES_REQUESTED** [review]

File: Curve-finance-suite/Pools-RampA/package.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    "@types/lru-cache": "^5.1.1",
+    "bignumber.js": "^9.0.1",
+    "forta-agent": "^0.0.32",
+    "forta-agent-tools": "^1.0.31",
+    "jest-when": "^3.4.2",
+    "lru-cache": "^6.0.0"
+  },
+  "devDependencies": {
+    "@types/jest": "^27.0.1",
+    "@types/nodemon": "^1.19.0",
```
[the remark this answers is not in the document]
Comment (e1Ru1o) [reply Curve-finance-suite/Pools-RampA/package.json:32]: Used in one of the scripts, check `start:dev`

**Review by mauricio1802 — APPROVED** [review]

**Review by RCantu92 — APPROVED** [review]

**Pull request merged**